libgeopm/include/geopm/Environment.hpp

```diff
@@ -116,16 +118,16 @@ namespace geopm
             int num_proc(void) const override;
             bool do_ctl_local(void) const override;
         protected:
-            void parse_environment(void);
-            bool is_set(const std::string &env_var) const;
-            std::string lookup(const std::string &env_var) const;
+            void GEOPM_PRIVATE parse_environment(void);
+            bool GEOPM_PRIVATE is_set(const std::string &env_var) const;
+            std::string GEOPM_PRIVATE lookup(const std::string &env_var) const;
```

Comment:
Would that make it impossible for implementers outside of libgeopm.so to call these functions when extending this class?

Maybe a non-issue for this particular class, but I don't know whether we want to hide `protected` symbols generally.